app/harvesters/ia_harvester.rb

```diff
@@ -0,0 +1,200 @@
+require 'json'
+
+##
+# A harvester implementation for Internet Archive
+#
+class IaHarvester
+  include Krikri::Harvester
+
+  DEFAULT_THREAD_COUNT = 10
+  DEFAULT_HARVEST_NAME = 'ia'
+  DEFAULT_MAX_RECORDS = 0
+
+  DOWNLOAD_BASE_URI = 'http://archive.org/download'
+
+  ##
+  # Initialize, and set default options as appropriate.
+  #
+  # @param opts [Hash] a hash of options as defined by {.expected_opts}
+  #
+  # @example
+  #
+  #    IaHarvester.new(
+  #      :uri => 'http://archive.org/advancedsearch.php?fl%5B%5D=identifier&output=json'
+  #      :ia => {:collections => ['bostonpubliclibrary', 'getty']}
+  #    )
+  #
+  # Accepts options passed as `:ia => opts`
+  #
+  # Options allowed are:
+  #
+  #   - collections: An array of collections keys to harvest
+  #   - threads:     The number of records to fetch asynchronously
+  #                  in a batch (default: 10)
+  #   - name:        See Krikri::Harvester#initialize.  (default: "ia")
+  #   - max_records: The maximum number of records to harvest
+  #                  0 means no limit (default 0)
+  #
+  def initialize(opts = {})
+    @opts = opts.fetch(:ia, {})
+
+    collections = @opts.fetch(:collections, [])
+    if collections.empty?
+      msg = ':collections option is required but missing'
+      Krikri::Logger.log(:error, msg)
+      fail msg
+    else
+      # other parameters are required for a successful search so
+      # we can assume we're appending to an existing query string
+      collection_qs = '&q=collection:(' +
+        collections.join('%20OR%20') +
+        ')'
+      opts[:uri] += collection_qs
+    end
+
+    super
+
+    @opts[:threads] ||= DEFAULT_THREAD_COUNT
+    @opts[:name] ||= DEFAULT_HARVEST_NAME
+    @opts[:max_records] ||= DEFAULT_MAX_RECORDS
+
+    @http = Krikri::AsyncUriGetter.new(opts: { follow_redirects: true })
+  end
+
+  ##
+  # @see Krikri::Harvester.expected_opts
+  def self.expected_opts
+    {
+      key: :ia,
+      opts: {
+        collections: { type: :string, multiple_ok: true, required: true },
+        threads: { type: :integer, required: false },
+        name: { type: :string, required: false },
```

Comment:
`:name` should not be included in `#expected_opts` (see [here](https://github.com/dpla/KriKri/blob/develop/lib/krikri/harvester.rb#L150-L181))